inc/ustring.bi

```diff
@@ -0,0 +1,681 @@
+END OPERATOR
+
+PRIVATE OPERATOR DWSTR.CAST () BYREF AS WSTRING       'returns the string data (same as **).
+  OPERATOR = *cast(WSTRING PTR, m_pBuffer)
+END OPERATOR
+
+
+#if sizeof(wstring) = 2
+PRIVATE OPERATOR DWSTR.[] (BYVAL nIndex AS ulong) byref AS USHORT
+'***********************************************************************************************
+' Returns the corresponding ASCII or Unicode integer representation of the character at the
+' zerobased position specified by the nIndex parameter. Can be used to change a value too.
+'***********************************************************************************************
+static zero as ushort                                 'fallback for nIndex outside valid data
+
+  IF nIndex > (m_BufferLen \ u_size) THEN
+    zero = &HFFFF
+    OPERATOR = zero                                   'return 0
+    exit operator
+  end if
+  
+  OPERATOR = *cast(USHORT ptr, m_pBuffer + (nIndex * u_size))
+END OPERATOR
+
+
+#else
+PRIVATE OPERATOR DWSTR.[] (BYVAL nIndex AS ulong) byref AS Ulong
+'***********************************************************************************************
+' Returns the corresponding ASCII or Unicode integer representation of the character at the
+' zerobased position specified by the nIndex parameter. Can be used to change a value too.
+'***********************************************************************************************
+static zero as ulong                                  'fallback for nIndex outside valid data
+
+  IF nIndex > (m_BufferLen \ u_size) THEN
+    zero = &HFFFFFFFF
+    OPERATOR = zero                                   'return 0
+    exit operator
+  end if
+  
+  OPERATOR = *cast(Ulong ptr, m_pBuffer + (nIndex * u_size))
+END OPERATOR
+#endif
+
+
+'***********************************************************************************************
+' Assigns new text to the DWSTR.
+'***********************************************************************************************
+
+
+PRIVATE OPERATOR DWSTR.Let (BYREF wszStr AS CONST WSTRING)
+  this.Clear
+  this.Add(wszStr)
+END OPERATOR
+
+PRIVATE OPERATOR DWSTR.Let (BYREF ansiStr AS STRING)
+  this.Clear
+  this.Add(ansiStr)
+END OPERATOR
+
+PRIVATE OPERATOR DWSTR.Let (BYREF pwszStr AS WSTRING PTR)
+  IF m_pBuffer = cast(ubyte ptr, pwszStr) THEN EXIT OPERATOR              'ignore self assign
+  this.Clear
+  IF pwszStr = 0 THEN EXIT OPERATOR
+  this.Add(*pwszStr)
+END OPERATOR
+
+PRIVATE OPERATOR DWSTR.Let (BYREF cws AS DWSTR)
+  IF m_pBuffer = cws.m_pBuffer THEN EXIT OPERATOR     'ignore cws = cws
+  this.Clear
+  this.Add(cws)
+END OPERATOR
+
+
+'***********************************************************************************************
+' Appends a string to the DWSTR
+'***********************************************************************************************
+
+
+PRIVATE OPERATOR DWSTR.+= (BYREF wszStr AS WSTRING)   'appends a wstring to the DWSTR
```

Comment:
This and many other parameters should be `const` otherwise won't take a `const` argument.